File: TypeScript/simple-cipher.ts
import { randomLetter, indexFromLetter, letterFromIndex } from "./helper/letters";

function* loopString(str: string) {
    let index = 0;

    while (true) {
        if(index == str.length) {
            index = 0;
        }

        yield str[index++].toLowerCase();
    }
}

export class SimpleCipher {
    public key: string = "";

    constructor(key: string ="") {
        this.key = key === "" ? this.generateKey() : key
    }

    private generateKey(): string {
        return Array.from({ length: 100 }, () => randomLetter().toLowerCase()).join('');
    }

    private transform(message: string, operation: '+' | '-'): string {
        const generator = loopString(this.key);  
        let output = "";

        message.split('').forEach(letter => {
            let a = letter;
            let b = generator.next().value!;
            output += letterFromIndex(indexFromLetter(a) + (operation === '+' ? 1 : -1) * indexFromLetter(b));
        });

        return output;
    }
    
    encode(message: string): string {
        return this.transform(message, '+');
    }
    
    decode(message: string): string {
        return this.transform(message, '-');
    }
}

const simpleCipher = new SimpleCipher('abc');
console.log(simpleCipher.decode("iboaqcnecbfcr"));

File: C#/Simple Cipher/SimpleCipher.cs
using System;
using System.Collections.Generic;
using System.Linq;

using _Helper;

public class Generator
{
    public static IEnumerable<char> LoopString(string str)
    {
        int index = 0;

        while (true)
        {
            if (index == str.Length)
            {
                index = 0;
            }

            yield return str[index++];
        }
    }
}

public class SimpleCipher
{
    private string _key = "";

    public SimpleCipher()
    {
        this._key = GenerateKey();
    }

    public SimpleCipher(string key)
    {
        this._key = key;
    }

    private string GenerateKey()
    {
        var randomLetters = new char[100].Select(c => Letters.GetRandomCharacter());
        return string.Join("", randomLetters);
    }

    public string Key
    {
        get
        {
            return this._key;
        }
    }

    private string Transform(string message, char operation)
    {
        var generator = Generator.LoopString(this._key);
        var enumerator = generator.GetEnumerator();

        string output = "";

        foreach (char c in message)
        {
            enumerator.MoveNext();
          
            output += Letters.LetterFromIndex(Letters.IndexFromLetter(c) + (operation == '+' ? 1 : -1) * Letters.IndexFromLetter(enumerator.Current));
        }

        return output;
    }

    public string Encode(string plaintext)
    {
        return this.Transform(plaintext, '+');
    }

    public string Decode(string ciphertext)
    {
        return this.Transform(ciphertext, '-');
    }
}